### DISCUSSION TITLE
Technical advice on scheduling selfmessages

### DISCUSSION THREAD

**Discussion by zazim13** [omnetpp/omnetpp#978] [Q&A] [answered]
Hi,
I am writing this issue more to ask for a technical advice. I am writing a server application and I would like to schedule an event in order to answer to a request from a user. But I would like my answer to contain some information on the received packet. I am facing two main issues:

* Requests are coming randomly, and I would like to answer each of them. Do I have to create a selfmessage for each potential request that I could receive ?
* And also is it possible to include received packet information in the selfmessage ? The goal being to include in the answer some information gotten from the request.

Sincerly
Hakim

**Comment by rhornig** [comment, marked as answer]
1. yes. you can create a self message for each incoming message with the required delay.
2. there are several ways:
- you can add the info to the self message in a field
- you can attach the incoming message to the self message as a context pointer
- you can reuse the incoming message as a self message by re-scheduling it again for yourself. Later you will be able to distinguish between normal incoming packets and re-scheduled (timer) messages based on the sender module (as the timer messages were sent by your own module = self messages)

**Comment by zazim13** [comment]
Thanks for your answer.
